src/compiler/core.ts

```diff
@@ -604,6 +604,31 @@ namespace ts {
         return result;
     }
 
+    export function groupBy<T>(array: ReadonlyArray<T>, predicate: (elem: T, index: number) => string): T[][] {
+        if (!array) {
+            return undefined;
+        }
+
+        const collection = createMap<T[]>();
+        let result: T[][];

```

Comment:
The return type promises a defined result -- but here `result` would be undefined if the input array was empty. You could just make the return type a `ReadonlyArray` and use `emptyArray` if you want to avoid allocating memory.